src/services/jsDoc.ts

```diff
@@ -88,15 +89,27 @@ namespace ts.JsDoc {
         return undefined;
     }
 
-    export function getAllJsDocCompletionEntries(): CompletionEntry[] {
-        return jsDocCompletionEntries || (jsDocCompletionEntries = ts.map(jsDocTagNames, tagName => {
-            return {
-                name: tagName,
-                kind: ScriptElementKind.keyword,
-                kindModifiers: "",
-                sortText: "0",
-            };
-        }));
+    export function getAllJsDocCompletionEntries(shouldAppendAtSign: boolean): CompletionEntry[] {

```

Comment:
i would split these into two functions, `getJSDocTagNameCompletions` and `getJSDocTagCompletions` and call them based on the context.